### PR TITLE
Document migration from provision.fmf to tmt

### PR DESCRIPTION
**Pull request by janhavlin**
Solves [TFT-767](https://issues.redhat.com/browse/TFT-767).

### CODE REVIEW

File: docs/questions.rst [hunk 1]
```diff
@@ -419,3 +419,37 @@ tmt example plan (L2 metadata):
...
+.. code-block:: yaml
+
+    provision:
+        how: virtual
+        hardware:
+            disk:
+                - size: ">10GiB"
+                - size: ">10GiB"
+                - size: ">10GiB"
+
```
Comment (thrix) [inline docs/questions.rst:453]: @janhavlin could we mention that this currently works only with Artemis provisioner, so really it will work only in Testing Farm?

Comment (janhavlin) [reply docs/questions.rst:453]: Sure, mentioning it sounds good. Just a question:

> this currently works only with Artemis provisioner

This part sounds right.

> so really it will work only in Testing Farm?

Is this really true? In the [`disk.fmf`](https://github.com/teemtee/tmt/blob/main/spec/hardware/disk.fmf) spec, it is mentioned that it is implemented by `tmt/steps/provision/artemis.py`, therefore it should work with both Testing Farm and `artemis` TMT plugin when using `tmt` directly?

Comment (happz) [reply docs/questions.rst:453]: I wouldn't mention Artemis by name. The support will change, this feature will be supported by more plugins, and nobody will recall updating this particular spot to include them. I believe the `See ...` note below is the right way, refer to a page dedicated to HW requirements, and we can extend it with "... for more details and plugin support.", for example.

Comment (happz) [reply docs/questions.rst:453]: > I wouldn't mention Artemis by name. The support will change, this feature will be supported by more plugins, and nobody will recall updating this particular spot to include them. I believe the `See ...` note below is the right way, refer to a page dedicated to HW requirements, and we can extend it with "... for more details and plugin support.", for example.

Edit: I wouldn't mention TF by name either...

Comment (thrix) [reply docs/questions.rst:453]: @happz @janhavlin so from discussion, let's link from this guide to the Testing Farm `Test Request` page:

https://docs.testing-farm.io/general/0.1/test-request.html

Comment (thrix) [reply docs/questions.rst:453]: Rather than adding note about that this is supported only there.

Comment (janhavlin) [reply docs/questions.rst:454]: Added a link to the TF docs. PTAL. Although I'm not sure how to link the TF docs if naming TF is not a good practice.

Comment (happz) [reply docs/questions.rst:454]: Mentioning TF in *specs* would be sort of vendor-lock-in-ish, it's always good to think whether it makes sense to point out a single tmt runner in such case. But here we are in a "how to migrate" part of tmt docs, and here it makes more sense as we are dealing with existing systems. No worries here.

I'd just maybe reword the sentence a bit: it's definitely possible to express the requirement, TF is mighty and can do anything, how about aligning it with the migration effort? Let's say "If you were using `provision.yaml` with Testing Farm, check out the TF docs on this HW requirement, https://..., for more details and how TF works with tmt metadata." I don't like the notion of "hey, this is possible with TF" - yep, that's true, but it will be possible with other systems one day. But when looking at it from the PoV of folks who were using the file, and want to migrate and use TF, which was the original motivation, singling TF out seems more justifiable to me.

Comment (janhavlin) [reply docs/questions.rst:454]: I see, that is indeed a bit better wording. One issue that I see with it is that currently, the TF docs page (https://docs.testing-farm.io/general/0.1/test-request.html) is a bit lacking with the explanation of "how TF works with tmt metadata" since currently it only describes how to express the same hardware specification with TF request syntax. That is also something that needs to improve as we already discussed it?

Comment (happz) [reply docs/questions.rst:454]: I guess so, yes, on TF side.

Comment (thrix) [reply docs/questions.rst:454]: @janhavlin waiting for docs update on TF side and resolving this one.

File: docs/questions.rst [hunk 2]
```diff
@@ -419,3 +419,44 @@ tmt example plan (L2 metadata):
...
+See the :ref:`/spec/hardware/disk` and :ref:`/spec/hardware/network`
+documentation for more details about these hardware specifications
+in tmt plans.
+
+If you were using ``provision.fmf`` with Testing Farm, check out
+the Testing Farm docs on this HW requirement,
+https://docs.testing-farm.io/general/0.1/test-request.html,
```
Comment (thrix) [inline docs/questions.rst:461]: @janhavlin please see how tmt developers write links to other pages, you can find example in this file

Comment (janhavlin) [reply docs/questions.rst:461]: You forgot to link it. :)

But the way I did it (just paste the link without any rst formatting) seems to be common, [example 1](https://github.com/teemtee/tmt/blob/main/docs/contribute.rst?plain=1#L254), [example 2](https://github.com/teemtee/tmt/blob/main/docs/examples.rst?plain=1#L414), [example 3](https://github.com/teemtee/tmt/blob/main/README.rst?plain=1#L540).

But yeah, the most common way is to use:
```
See `TF docs`__.

__ link
```

Comment (janhavlin) [reply docs/questions.rst]: Fixed, yeah, looks way better.

File: docs/questions.rst [hunk 3]
```diff
@@ -419,3 +419,45 @@ tmt example plan (L2 metadata):
...
 See the :ref:`/spec/plans/discover/fmf` plugin documentation for
 more details.
+
+
+Migrating provision.fmf
+------------------------------------------------------------------
+
+The ``provision.fmf`` file is used to specify storage and network
+devices. In this migration, the contents of the ``provision.fmf``
+file is moved to the ``provision`` step under ``hardware``
```
Comment (thrix) [inline docs/questions.rst:429]:
```suggestion
file are moved to the ``provision`` step under ``hardware``
```

**Review by thrix — APPROVED** [review]

**Review by happz — APPROVED** [review]
